src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/operations/streaming_endpoint.py

```diff
@@ -64,6 +64,41 @@ def remove_akamai_access_control(client, account_name, resource_group_name, stre
     return client.update(resource_group_name, account_name, streaming_endpoint_name, streaming_endpoint)
 
 
+def update_streaming_endpoint(client, resource_group_name, account_name, streaming_endpoint_name,  # pylint: disable=too-many-locals
+                              tags=None, cross_domain_policy=None, ips=None, client_access_policy=None,
+                              description=None, max_cache_age=None,
+                              cdn_provider=None, cdn_profile=None, custom_host_names=None):
+    from azure.mgmt.media.models import (IPAccessControl, StreamingEndpointAccessControl)
+
+    streaming_endpoint = client.get(resource_group_name, account_name, streaming_endpoint_name)
```

Comment:
After adding support for generic updates, you will have to check if the instance is None and raise a CLIError